lib/gitsh/environment.rb

```diff
@@ -50,6 +50,14 @@ def fetch(key, force_default_git_command = false, &block)
       raise Gitsh::UnsetVariableError, "Variable '#{key}' is not set"
     end
 
+    def available_variables
+      (
+        magic_variables.available_variables +
+        variables.keys +
+        repo.available_config_variables
+      ).sort.uniq
```

Comment:
For the most micro of micro optimizations, you could `uniq.sort` -- this will cause sort to have a smaller list to work through.